WordPress/src/main/java/org/wordpress/android/ui/activitylog/RewindStatusService.kt

```diff
@@ -62,59 +64,79 @@ constructor(
         site = null
     }
 
+    private fun reloadRewindStatus(): Boolean {
+        site?.let {
+            val state = activityLogStore.getRewindStatusForSite(it)
+            state?.let {
+                updateRewindStatus(state)
+                return true
+            }
+        }
+        return false
+    }
+
     private fun updateRewindStatus(rewindStatus: RewindStatusModel?) {
         mutableRewindAvailable.postValue(
                 rewindStatus != null &&
                         rewindStatus.state == ACTIVE &&
                         rewindStatus.rewind?.status != RUNNING
         )
-        mutableRewindState.postValue(rewindStatus?.rewind)
+        val rewind = rewindStatus?.rewind
+        mutableRewindState.postValue(rewind)
+        if (rewind != null) {
+            rewind.rewindId?.let { rewindId ->
+                updateRewindProgress(rewindId, rewind.progress, rewind.status, rewind.reason)
+            }
+            if (rewind.status != RUNNING) {
+                workerController.cancelWorker()
+            }
+        }
     }
 
     @Subscribe(threadMode = ThreadMode.BACKGROUND)
     @SuppressWarnings("unused")
     fun onRewindStatusFetched(event: OnRewindStatusFetched) {
+        mutableRewindStatusFetchError.postValue(event.error)
         if (event.isError) {
-            event.error?.let {
-                mutableRewindStatusFetchError.postValue(it)
-            }
             workerController.cancelWorker()
-        } else {
-            mutableRewindStatusFetchError.postValue(null)
-        }
-        site?.let {
-            val rewindStatusForSite = activityLogStore.getRewindStatusForSite(it)
-            rewindStatusForSite?.let {
-                updateRewindStatus(it)
-                it.rewind?.let {
-                    if (it.status != RUNNING) {
-                        workerController.cancelWorker()
-                    }
-                }
-            }
         }
+        reloadRewindStatus()
     }
 
     @Subscribe(threadMode = BACKGROUND)
     @SuppressWarnings("unused")
     fun onRewind(event: OnRewind) {
+        mutableRewindError.postValue(event.error)
```

Comment:
Shouldn't this be triggered only when there is an error?

Reply:
the idea here was to post `null` when there is no error. It's a bit tricky when we should clean these states